### ISSUE TITLE
"Non-exhaustive patterns in function checkCmd" error

### ISSUE THREAD

**Issue by antoniszymanski** [koalaman/shellcheck#3479]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"): `v0.11.0` and `v0.11.0-79-g764802b`
- [YES] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [NO] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/bash
profile=$(source "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")/profile.sh")
```

Checked using this command:
```bash
shellcheck --external-sources $FILE
```

#### Here's what shellcheck currently says:
`v0.11.0`:
```
shellcheck: src/ShellCheck/Analytics.hs:4009:5-93: Non-exhaustive patterns in function checkCmd
```
`v0.11.0-79-g764802b`:
```
shellcheck: src/ShellCheck/Analytics.hs:4031:5-93: Non-exhaustive patterns in function checkCmd
```

#### Here's what I wanted or expected to see:
No errors.

**Comment by antoniszymanski** [comment]
But for this snippet ShellCheck works perfectly fine:
```bash
#!/usr/bin/bash
profile=$(bash "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")/profile.sh")
```

**Comment by antoniszymanski** [comment]
And without the `--external-sources` flag.
